staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -158,7 +172,12 @@ func BackoffUntil(f func(), backoff BackoffManager, sliding bool, stopCh <-chan
 		}()
 
 		if sliding {
-			t = backoff.Backoff()
+			interval = delayFn()
+		}
+		if t == nil {
+			t = timerFn(interval)
+		} else {
+			t.Reset(interval)
```

Comment:
https://pkg.go.dev/time#Timer.Reset

> For a Timer created with NewTimer, Reset should be invoked only on stopped or expired timers with drained channels.

does this meet this requirement?

Reply:
To get to this point in the function, we had to have waited on T's channel at least once.  All other paths exit the function.

Note: you *don't* have to drain a channel for it to get garbage collected, you only have to Stop it (that's a VERY subtle thing that was tripping up people before).